### PR TITLE
1134 saveprompt in other components

### PR DESCRIPTION
**Pull request by ahjyrkia**
closes https://github.com/HSLdevcom/jore-map-ui/issues/1134

savePrompt added to lineView, routePathView and LineHeaderTable

### CODE REVIEW

File: src/components/sidebar/lineView/LineHeaderTable.tsx [hunk 1]
```diff
@@ -146,6 +147,44 @@ class LineHeaderTable extends React.Component<ILineHeaderListProps, ILineHeaderS
...
+    private showSavePrompt = () => {
+        const confirmStore = this.props.confirmStore;
+        const oldLineHeaders = this.props.lineHeaderMassEditStore!.oldLineHeaders!;
+        const currentLineHeaders = this.props.lineHeaderMassEditStore!.currentLineHeaders!;
+        const saveModels: ISaveModel[] = [];
+
+        currentLineHeaders.forEach((currentLineHeader: ILineHeader) => {
+            !currentLineHeader.originalStartDate
+                ? saveModels.push({
+                      newData: currentLineHeader,
```
Comment (Divinare) [inline src/components/sidebar/lineView/LineHeaderTable.tsx]: Add subtopic: currentLineHeader.lineNameFi

**Comment by Divinare** [comment]
Testing:

These should be localized:

LineView:
JOUKKOLIIKENNELAJI
01 --> 02

LINJAN KORVAAVA TYYPPI
--> 1

RoutePathView:
POIKKEUSREITTI
0 --> 1

ALSO, routePathNodes could be rendered with putting them into ISaveModel objects and passing an array of ISaveModel[] to SavePrompt.

End result would be something like:

Reitinsuunta 0033

Nimi suomeksi Nimi ruotsiksi
a -> b

---

Reitinsuunnan linkit ja solmut muuttuneet

---

Pysäkki Myllynkivenkuja
Ajantasauspysäkki
Kyllä -> Ei

Pysäkki Raappavuorenkuja
Pysäkki ei käytössä
Kyllä -> Ei

Note: "Reitinsuunnan linkit ja solmut muuttuneet" would be a generic notification that is displayed always if currentRoutePathLinks !== oldRoutePathLinks. Then under that, there would be a list of routePath stops (modified and added). Push them into ISaveModel[]

**Comment by ahjyrkia** [comment]
Requested changes done and new task created https://github.com/HSLdevcom/jore-map-ui/issues/1192

**Comment by Divinare** [comment]
lineHeaderMassEdit

Bug: When creating new lineHeader, it shows rows  with empty old and new attribute "     "  -> "      "

How to fix this: maybe check if both new and old attribute are empty, dont display that row.

**Comment by Divinare** [comment]
oldRoutePathLinks

**BUG 1**: if changing ONLY hastus of a routePathLinkNode, text "reitinlinkit ovat muuttuneet" doesn't show up

**BUG 2**: (shown in the picture): text of "Vanhat reitinsuunnanlinkit -> Uudet reitinsuunnanlinkit" shows under "Poikkeusreitti", should show in its own row under "Reitinlinkit ja solmut"
   * Also the text should be "Vanhat reitinsuunnan linkit ja solmut -> uudet reitinsuunnan linkit ja solmut"

**Comment by Divinare** [comment]
lineHeader topic

Maybe would be more clear if subtopics where like:
Linjan otsikko: 45
Linjan otsikko: 4a

**Comment by ahjyrkia** [comment]
Requested changes done and merged with develop

**Pull request merged**